gpytorch/lazy/lazy_tensor.py

```diff
@@ -732,6 +734,217 @@ def add_jitter(self, jitter_val=1e-3):
         diag = torch.tensor(jitter_val, dtype=self.dtype, device=self.device)
         return self.add_diag(diag)
 
+    def cat_rows(self, cross_mat, new_mat, generate_roots=True):
```

Comment:
Maybe this method should take in a **root_decomp_args that get passed into `root_decomposition` - if `generate_roots` is true?